packages/plugin-vue/src/handleHotUpdate.ts

```diff
@@ -88,7 +88,11 @@ export async function handleHotUpdate({
     const next = nextStyles[i]
     if (!prev || !isEqualBlock(prev, next)) {
       didUpdateStyle = true
-      const mod = modules.find((m) => m.url.includes(`type=style&index=${i}`))
+      const mod = modules.find(
+        (m) =>
+          m.url.includes(`type=style&index=${i}`) &&
+          m.url.endsWith(`.${next.lang || 'css'}`)
```

Comment:
**question:** 

> "[...] changing the lang attribute of the style tag will not cause the preprocessor to be updated. [...]"

Why should this work? You just shrink the condition from `includes` to `includes + endsWith` 🤔
Could you explain what is happening here?

Reply:
When the lang is changed, the corresponding module will be found through the following line of code, but the url of the module still contains the previous lang. The css plugin will parse the url of this module to select the preprocessor. 
https://github.com/vitejs/vite/blob/b5ab77d878d7bdc5c68dbbf144395f2a590f7d99/packages/plugin-vue/src/handleHotUpdate.ts#L91
After adding endsWith, a new module will be generated with new lang in the url.